internal/collection/workflow.go

```diff
@@ -30,18 +29,31 @@ type ProcessingWorkflowRequest struct {
 	// an existing collection in retries.
 	CollectionID uint
 
-	// Captured by the watcher, the event contains information about the
-	// incoming dataset.
-	Event *watcher.BlobEvent
+	WatcherName string
+
+	PipelineName string
+
+	RetentionPeriod *time.Duration
```

Comment:
Most of the attributes are self-explanatory but we could clarify this one, it's always hard to remember the purpose of pointers, i.e. in this case IIRC, nil means no deletion whereas 0 is immediate deletion and gt 0 is scheduled deletion.